pkg/virt-handler/vm.go

```diff
@@ -2166,6 +2166,19 @@ func (d *VirtualMachineController) checkVolumesForMigration(vmi *v1.VirtualMachi
 				return true, fmt.Errorf("cannot migrate VMI with non-shared HostDisk")
 			}
 		} else {
+			isVolumeUsedByReadOnlyDisk := false
+			for _, disk := range vmi.Spec.Domain.Devices.Disks {
```

Comment:
Maybe this could be moved out of `for _, volume := range vmi.Spec.Volumes {` and aggregated into a map.

Reply:
What is the motivation behind it?
At the moment this code runs only under certain conditions (volumes exist that are not PVCs or DVs, etc'), if we move it out of the `for` loop then this map would've been, in some cases, created for nothing.

Reply:
Thanks make sense now.